File: packages/cloud-adapters/storage/interface.go
// Package storage defines the cloud-adapter storage axis interface.
// Bare-metal impl: MinIO. Cloud impls: S3 / GCS / Azure Blob.
package storage

import "context"

type Adapter interface {
	Put(ctx context.Context, key string, body []byte) error
	Get(ctx context.Context, key string) ([]byte, error)
	Delete(ctx context.Context, key string) error
	List(ctx context.Context, prefix string) ([]string, error)
}

File: packages/cloud-adapters/storage/interface.ts
// Storage adapter axis — object-storage primitives the platform uses for
// tenant uploads / exports / cached blobs. Bare-metal is MinIO; cloud
// impls map to S3 / GCS / Azure Blob.

export interface StorageAdapter {
  put(key: string, body: Uint8Array): Promise<void>;
  get(key: string): Promise<Uint8Array>;
  delete(key: string): Promise<void>;
  list(prefix: string): Promise<string[]>;
}
